Error Codes & Debugging
===========================

I will try compile as many different error codes and their explanations here, as well as potential causes and solutions to deal with them as and when they pop up. If there are any you wish to add, please message me on my discord "theonetruenerd".

Runtime Errors
--------------------

- :py:func:`0xa0220001` : No memory
- :py:func:`0xa1230002` : Inserting identifier failed
- :py:func:`0xa1230003` : Identifier not found
- :py:func:`0xa2230004` : L-value not a number
- :py:func:`0xa2230005` : R-value not a number
- :py:func:`0xa1230006` : Not an identifier
- :py:func:`0xa1220007` : Unrecognized token
- :py:func:`0xa2230008` : R-value not bound
- :py:func:`0xa2230009` : Bad number
- :py:func:`0xa123000a` : Bad tree
- :py:func:`0xa123000b` : Invalid entry
- :py:func:`0xa122000c` : Function identifier is protected
- :py:func:`0xa223000d` : Underspecified
- :py:func:`0xa2230037` : Overspecified
- :py:func:`0xa123000e` : Setting value failed
- :py:func:`0xa123000f` : Function identifier not found
- :py:func:`0xa1230010` : Bindings not found
- :py:func:`0xa1230011` : Temporary variable not found
- :py:func:`0xa1230012` : Unknown function type
- :py:func:`0xa2230013` : Unable to find file
- :py:func:`0xa1230014` : Type mismatch
- :py:func:`0xa2230015` : Bad L-value
- :py:func:`0xa2230016` : Bad R-value
- :py:func:`0xa2220017` : Unrecognised type
- :py:func:`0xa1230018` : Bad memory type
- :py:func:`0xa1230019` : Array reference out of bound
- :py:func:`0xa123001a` : Bad array identifier type
- :py:func:`0xa123001b` : Tag insert failed
- :py:func:`0xa123001c` : Dynamic memory identifier not bound
- :py:func:`0xa123001d` : Tag identifier not bound
- :py:func:`0xa123001e` : Structure reference out of bound
- :py:func:`0xa123001f` : Bad tag identifier type
- :py:func:`0xa1230020` : L-value is not a structure identifier
- :py:func:`0xa1230021` : L-value is not an array identifier
- :py:func:`0xa1230022` : Failed to lookup tag identifier in tag table
- :py:func:`0xa1230023` : Signal break
- :py:func:`0xa1230024` : Copy out of bound
- :py:func:`0xa1230025` : Signal return
- :py:func:`0xa2230026` : Array size is not an integer
- :py:func:`0xa1230027` : Failed to copy tag table
- :py:func:`0xa1230029` : Function has not been defined
- :py:func:`0xa123002a` : Unable to enter nesting level
- :py:func:`0xa123002b` : Unable to exit nesting level
- :py:func:`0xa122002c` : No context
- :py:func:`0xa123002d` : Failed to read file
- :py:func:`0xa123002e` : Failed to create timer
- :py:func:`0xa123002f` : Failed to set timer
- :py:func:`0xa1230030` : Failed to wait timer
- :py:func:`0xa1230031` : Failed to create event
- :py:func:`0xa1230032` : Failed to set event
- :py:func:`0xa1230033` : Failed to wait event
- :py:func:`0xa1230034` : Bad argument
- :py:func:`0xa2230035` : Syntax error
- :py:func:`0xa2230036` : Integer divide by zero
- :py:func:`0xa2230038` : Returning address of local variable or temporary
- :py:func:`0xa223003a` : Unable to find file
- :py:func:`0xa223003b` : File not updatable
- :py:func:`0xa223003c` : Recursive call
- :py:func:`0xa223003d` : Failed to wait for thread(s)
- :py:func:`0xa223003e` : Time-out interval elapsed
- :py:func:`0xa2220044` : Automation type not supported
- :py:func:`0xa1230046` : Bad argument parameter
- :py:func:`0xa123004d` : Sequence property not found
- :py:func:`0xa123004e` : int64 not supported
